src/common/reduction/matching/GapMatcher.h

```diff
@@ -36,6 +40,30 @@ class GapMatcher : public AbstractMatcher {
   /// \brief print configuration of GapMatcher
   std::string config(const std::string &prepend) const override;
 
+  void reset_stats() {
+    Stats.SpanTooLarge = 0;
+    Stats.DiscardedSpanTooLarge = 0;
+    Stats.SplitSpanTooLarge = 0;
+  }
+
+  struct Stats {
+    uint16_t SpanTooLarge{0};
+    uint16_t DiscardedSpanTooLarge{0};
+    uint16_t SplitSpanTooLarge{0};
+  } Stats;
+
 private:
+private:
+  void split_and_stash_event(Event evt);
```

Comment:
I'd like to add doxygen comments, generally to all methods in .h files (I might forget to do this too, but I
think this is one of the most important things to do)
/// \brief ...
/// \param cluster ...
...
/// \return if method returns anything